### PR TITLE
Create Existing comment component

### PR DESCRIPTION
**Pull request by ArmanyaDalmia**
## Notion ticket link

[Create existing comment component](https://www.notion.so/uwblueprintexecs/a658c933d18c48d4b43fd99bda2bd3b4?v=4a47edc475b34bb99b25ff1fae2a5507&p=db62b18f0f9f4422a833567c45883520)

## Implementation description

- Created gql mutation for updating user
- Created Existing comment component
image

## Steps to test

1. Confirm pressing reply opens a WIPComment component (only when the Comments button is toggled) and pressing cancel on the WIPComment component closes it
2. Confirm pressing respond updates the resolved value for the comment in the db and updates the value of the comments state var

## What should reviewers focus on?

- That the functionality is as desired

## Checklist

- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] For backend changes, I have run the appropriate linters: `docker exec -it planet-read_py-backend_1 /bin/bash -c "black . && isort --profile black ."` and I have generated new migrations: `flask db migrate -m "<your message>"`
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/review/ExistingComment.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import { useMutation } from "@apollo/client";
+import { Flex, Box, Button } from "@chakra-ui/react";
+import AuthContext from "../../contexts/AuthContext";
+import WIPComment from "./WIPComment";
+import {
+  UPDATE_COMMENT_BY_ID,
+  UpdateCommentResponse,
+} from "../../APIClients/mutations/CommentMutations";
+
+export type ExistingCommentProps = {
```
Comment (gaoxk) [inline frontend/src/components/review/ExistingComment.tsx:11]: dont forget about displaying the time! 
image

File: frontend/src/components/review/ExistingComment.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  const name = `
+    ${authenticatedUser!!.firstName}
+    ${authenticatedUser!!.lastName}`;
+
+  return (
+    <Flex>
+      <Box backgroundColor="gray.300" float="right" width="250px">
+        <p>{name}</p>
+        {content}
+        <Button onClick={() => setReply(true)}>Reply</Button>
```
Comment (gaoxk) [inline frontend/src/components/review/ExistingComment.tsx:65]: some thoughts about styling (that you might already be implementing): you'll want to add a new button variant in chakra ui for styling the Reply and Resolve buttons (eg theme/components/Button.ts)

File: frontend/src/components/review/ExistingComment.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        <Button onClick={resolveExistingComment}>Resolve</Button>
+      </Box>
+      {reply === true && (
```
Comment (gaoxk) [inline frontend/src/components/review/ExistingComment.tsx:68]: nit: i think you can simplify to `{reply && (`

**Review by gaoxk — CHANGES_REQUESTED** [review]
drive by review!

File: frontend/src/components/review/ExistingComment.tsx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+      style={{
+        borderRadius: 8,
+        width: "320px",
+      }}
+      backgroundColor="white"
+    >
+      <b>Line {lineIndex}</b>
+      <Flex justify="space-between">
+        <p>{name}</p>
+        <p>{time}</p>
```
Comment (gaoxk) [inline frontend/src/components/review/ExistingComment.tsx:79]: fyi: comments panel is not fetching time right now in `CommentsQueries.ts`
```
const COMMENT_FIELDS = `
    id
    commentIndex
    content
    storyTranslationContentId
    `;
```
Are you testing this component by like so in the comments panel?
```
const CommentsList = () => {
         ....
          <ExistingComment
              key={comment.id}
              id={comment.id}
              resolved={comment.resolved}
              content={comment.content}
              time={comment.time}
              commentStoryTranslationContentId={
                commentStoryTranslationContentId
              }
              setCommentLine={setCommentLine}
              lineIndex={comment.commentIndex}
            />
          ))}
```
comments panel looks like this for me rn (time is missing)
image
Based on the mocks, the time should also be on the same line as the name. Feel free to wrap these two in a `Flex` and `justifyContent: space-between` or something along those lines

File: frontend/src/theme/components/Button.ts [hunk 5]
```diff
@@ -51,6 +51,12 @@ const Button = {
       right: "10px",
       width: "50px",
     },
+    text: {
```
Comment (gaoxk) [inline frontend/src/theme/components/Button.ts:54]: hmmm maybe a better name is `label` as it's called on figma?

**Review by gaoxk — CHANGES_REQUESTED** [review]
It's coming along! Btw, I don't mind if you include the code use of the ExistingComment component in the comments panel. Makes it easier for me to test too lmao

**Comment by gaoxk** [comment]
one more fyi - the reply button did nothing for me when I used the `ExistingComment` component in the `CommentPanel`

File: frontend/src/theme/components/Button.ts [hunk 6]
```diff
@@ -51,6 +51,14 @@ const Button = {
       right: "10px",
       width: "50px",
     },
+    label: {
```
Comment (gaoxk) [inline frontend/src/theme/components/Button.ts:54]: nit: sort these attributes alphabetically

File: frontend/src/components/review/ExistingComment.tsx [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+      </Text>
+      <Flex>
+        <Button onClick={() => setReply(1)} variant="label">
+          Reply
+        </Button>
+        <Button onClick={resolveExistingComment} variant="label">
+          Resolve
+        </Button>
+      </Flex>
+      {reply > 0 && lineIndex > -1 && (
```
Comment (gaoxk) [inline frontend/src/components/review/ExistingComment.tsx:92]: when is `lineIndex == -1`?

File: frontend/src/components/review/ExistingComment.tsx [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+    }
+  };
+
+  const name = `
+    ${authenticatedUser!!.firstName}
+    ${authenticatedUser!!.lastName}`;
+
+  return (
+    <Flex
+      border="1px solid"
```
Comment (gaoxk) [inline frontend/src/components/review/ExistingComment.tsx:64]: nit: alphabetize these props

File: frontend/src/components/review/ExistingComment.tsx [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+      borderColor="blue.50"
+      padding="14px 15px"
+      direction="column"
+      style={{
```
Comment (gaoxk) [inline frontend/src/components/review/ExistingComment.tsx:68]: why is the `style` prop needed? Can `borderRadius` and `width` be set as a usual chakra ui prop?

File: frontend/src/theme/components/Button.ts (hunk 6 above)
Comment (gaoxk) [inline frontend/src/theme/components/Button.ts:54]: nit: `label` is a bit of a vague line, what about `commentLabel`

File: frontend/src/components/review/ExistingComment.tsx [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+  const resolveExistingComment = async () => {
+    try {
+      if (!resolved) {
+        const commentData = {
+          id,
+          resolved: true,
+          content,
+        };
+
+        const result = await updateComment({
```
Comment (gaoxk) [inline frontend/src/components/review/ExistingComment.tsx:49]: Nothing is being done with this result (getting a build error for it)

File: frontend/src/components/review/ExistingComment.tsx [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+  time,
+  commentStoryTranslationContentId,
+  lineIndex,
+}: ExistingCommentProps) => {
+  const handleError = (errorMessage: string) => {
+    // eslint-disable-next-line no-alert
+    alert(errorMessage);
+  };
+
+  const [reply, setReply] = useState(0);
```
Comment (gaoxk) [inline frontend/src/components/review/ExistingComment.tsx:33]: I'm trying to wrap my head around `reply` and `commentLine`: are you treating `reply` as a boolean/flag for displaying a `WIPComment`?

**Review by gaoxk — CHANGES_REQUESTED** [review]

**Comment by gaoxk** [comment]
The comment button seems to always be disabled for me 
image

**Comment by ArmanyaDalmia** [comment]
> The comment button seems to always be disabled for me image

image
I'm able to reply to the comment, so I have no idea why the button is disabled for you

**Comment by gaoxk** [comment]
Tested it again and this seems to be the core issue:
1. click reply
2. enter text into the WIPComment component and hit comment
3. get the following error
image
this is the network response from the createComment mutation:
```
{"errors":[{"message":"'NoneType' object has no attribute 'stage'","locations":[{"line":2,"column":3}],"path":["createComment"]}],"data":{"createComment":null}}
```
4. I run into disabling issues after that

**Review by gaoxk — APPROVED** [review]
🚢 🚢 🚢 🚢

**Pull request merged**